cpp/oneapi/dal/table/detail/sparse_block.hpp

```diff
@@ -0,0 +1,42 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/table/common.hpp"
+
+namespace oneapi::dal::detail {
+namespace v1 {
+
+/// Type of csr indexing, currently it's implemented only `one_based` indexing.
+enum class csr_indexing { zero_based, one_based };
+
+/// @tparam T The type of data values in block.
+///           Sparse block supports at least :expr:`float`, :expr:`double`, and :expr:`std::int32_t` types of :literal:`T`.
+template <typename T>
+struct sparse_block {
```

Comment:
csr_block?

Reply:
I suppose it's better to save 'sparse_block' name because inside we have the same naming of the struct. This struct might be reused in future (for example compressed sparse column)

Reply:
'sparse' was changed to 'csr' in whole PR